app/models/project/Page.scala

```diff
@@ -120,6 +123,44 @@ case class Page(override val id: Option[Int] = None,
 
 object Page {
 
+  private object ExternalLinkResolver {
+
+    class Factory(config: OreConfig) extends LinkResolverFactory {
+      override def getAfterDependents() = null
+      override def getBeforeDependents() = null
+      override def affectsGlobalScope() = false
+      override def create(context: NodeRendererContext) = new ExternalLinkResolver(this.config)
+    }
+
+  }
+
+  private class ExternalLinkResolver(config: OreConfig) extends LinkResolver {
+    override def resolveLink(node: Node, context: NodeRendererContext, link: ResolvedLink) = {
+      if (link.getLinkType.equals(LinkType.IMAGE) || node.isInstanceOf[MailLink]) {
+        link
+      } else {
+        link.withStatus(LinkStatus.VALID).withUrl(wrapExternal(link.getUrl))
+      }
+    }
+
+    private def wrapExternal(urlString: String) = {
+      try {
+        val uri = new URI(urlString)
+        if (uri.getScheme == null || "http" == uri.getScheme || "https" == uri.getScheme) {
+          val host = uri.getHost
+          val trustedUrlHost = this.config.app.getString("trustedUrlHost").get
+          if (host == null || host.endsWith(trustedUrlHost)) {
+            urlString
+          } else {
+            controllers.routes.Application.linkOut(urlString).toString()
+          }
+        } else {
+          urlString
+        }
+      } catch { case _: URISyntaxException => urlString }
```

Comment:
Don't default to untrusted user input for a failure condition.

Your current implementation is trivially bypassable simply by appending `?a""` to the end of the link URL.